rcl/src/rcl/client.c

```diff
@@ -278,14 +278,15 @@ rcl_take_response(
   return RCL_RET_OK;
 }
 
-bool rcl_client_is_valid(const rcl_client_t * client)
+bool rcl_client_is_valid(const rcl_client_t * client, const rcl_allocator_t * allocator)
 {
   const rcl_client_options_t * options;
-  RCL_CHECK_ARGUMENT_FOR_NULL(
-    client, false, rcl_get_default_allocator());
+  const rcl_allocator_t alloc = allocator ? *allocator : rcl_get_default_allocator();
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&alloc, "client's allocator is invalid", return false);
```

Comment:
Not sure how we should phrase it but when reading `client's allocator is invalid` in my console, I understand that the allocator tied to my client is invalid (`options->allocator`) and not the one passed to the validation function.
We could either call the macro without an error message or clarify the message to make it obvious which allocator is invalid

Reply:
@mikaelarguedas that's a good point. Should I just mimic the message from #173?

Reply:
Or something like "argument alloc is invalid" or "passed allocator is invalid", as "invalid allocator" doesn't really disambiguate IMO.

Reply:
But so why aren't we using the client's allocator?

Reply:
Yeah that's a bit hairy, here we need to confirm that the client and client->options structure are valid before using the allocator stored in it. The allocator passed to this function is only used to allocate error messages if something goes wrong when checking the validity of the fields of the client.
Maybe a less confusing way would be to rename the parameter to be "error_message_allocator" this way we know it's not the client's allocator and it won't be used to allocate memory within client's normal operation mode.